sdk/core/core-client-rest/src/urlHelpers.ts

```diff
@@ -94,13 +190,17 @@ export function buildBaseUrl(endpoint: string, options: RequestParameters): stri
 
 function buildRoutePath(
   routePath: string,
-  pathParameters: string[],
+  pathParameters: (string | PathParameterWithOptions)[],
   options: RequestParameters = {},
-) {
+): string {
   for (const pathParam of pathParameters) {
-    let value = pathParam;
-    if (!options.skipUrlEncoding) {
-      value = encodeURIComponent(pathParam);
+    const allowReserved = isPathParameterWithOptions(pathParam)
+      ? (pathParam.allowReserved ?? false)
+      : false;
+    let value = isPathParameterWithOptions(pathParam) ? pathParam.value : pathParam;
```

Comment:
what do we do if some JS-only consumer were to pass `{value: undefined}`? 

I think we'd mistake it for a literal and get something like 
![image](https://github.com/user-attachments/assets/48d767b5-53c9-4c34-ae1b-a0459b34a3c9)

Maybe not worth guarding against, but since we're already doing the work to interrogate the type of the param, might be useful to throw/warn

Reply:
I made an update here to handle this scenario a bit more gracefully and to prevent any people being broken, on the off chance that they're passing undefined as a path param for some reason. The `undefined` will now just show up in the path, matching the existing behavior. Do you think that's enough or would you prefer it if we threw? I guess we would throw in both situations (undefined being passed directly without a wrapper or with `{value: undefined}` like your example).

Reply:
I am fine with that solution to start, we can consider stronger validation later if we see confusion/runtime issues